src/utils/avatar.js

```diff
@@ -389,7 +389,12 @@ export class UploadedAvatarURL extends AvatarURL {
          */
       result = new URL(this._standardUrl);
 
-      result.pathname = result.pathname.replace(/\.png$/, '-medium.png');
+      result.pathname = result.pathname.replace(
+        // `.png` is optional: s3 uploads don't use it, local ones do.
+        // See TODO in zulip/zulip@c03615b98.
+        /(?:\.png)?$/,
```

Comment:
Why the non-capturing group here? Is there a reason `/(\.png)?$/` wouldn't work?

Reply:
I tend to use non-capturing groups when I don't intend to read the captured value on the match object I get from executing the pattern.
- It has seemed neater to me to get rid of unused (and often never-useful) bits of data there.
- Also, the signal that we don't intend to use the captured value can help make patterns easier to read, I think. When I use a capturing group, I'm foreshadowing that I'll care about the value later on; when I use a non-capturing group, its role is clearer as just an instrument to say "treat this sequence of items together". I guess I'd prefer if the capturing-group syntax were the one that required extra characters. 🙂

What do you think?